test_stack.py

```diff
@@ -113,3 +113,42 @@ def test_quiet():
     assert eval_program("push 3; push 5; quiet sub") == [3, 5, -2]
     assert eval_program("push 3; push 5; quiet div") == [3, 5, 0.6]
     assert eval_program("push 3; push 5; quiet pow") == [3, 5, 3**5]
+
+
+def test_negation():
+    assert eval_program("push 1; not;") == [0]
+    assert eval_program("push -1; not;") == [0]
+    assert eval_program("push 0; not;") == [1]
+    assert eval_program("push 3; push 7; push 0; not;") == [3, 7, 1]
+    assert eval_program("push 1; quiet not;") == [1, 0]
+    assert eval_program("push 0; quiet not;") == [0, 1]
+    with pytest.raises(IndexError):
+        # Cannot negate top element since there isn't a top element
+        assert eval_program('not')
+
+
+def test_equality():
+    assert eval_program("push 0; push 0; eq") == [1]
+    assert eval_program("push 1.0; push 1.0; =") == [1]
+    assert eval_program("push 2.0; push 2; =") == [1]
+    assert eval_program("push -1; push -1; eq") == [1]
+    assert eval_program("push 3; push 5; eq") == [0]
+    assert eval_program("push 4; push 4; quiet eq") == [4, 4, 1]
+    assert eval_program("push 3; push 5; quiet eq") == [3, 5, 0]
+    with pytest.raises(IndexError):
+        assert eval_program("eq")
+    with pytest.raises(IndexError):
+        assert eval_program("push 1; eq")
+
+def test_inequality():
+    assert eval_program("push 5; push 7; lt") == [1]
+    assert eval_program("push 5; push 7; le") == [1]
+    assert eval_program("push 3; push 3; le") == [1]
+    assert eval_program("push -1; push 0; quiet <") == [-1, 0, 1]
+    assert eval_program("push 5.1; push 5.0; quiet >=") == [5.1, 5.0, 1]
+    assert eval_program("push -3; push 27; <=") == [1]
+    assert eval_program("push 7; push 5; gt") == [1]
+    assert eval_program("push 7; push 5; ge") == [1]
+    assert eval_program("push 3; push 3; ge") == [1]
+    with pytest.raises(IndexError):
+        assert eval_program(">")
```

Comment:
You should also have a test for `IndexError` with only a single operand, like you do in `test_equality`.